singer-io/tap-mambu
# Update custom field schema

# Description of change
Update transform and schema for custom fields to improve flattening that happens in some destinations(Postgres in particular). 

New schema for custom fields:

```json
"custom_fields": {
            "anyOf": [{
                    "type": "array",
                    "items": {
                        "type": "object",
                        "additionalProperties": false,
                        "properties": {
                            "field_set_id": {
                                "type": ["null", "string"]
                            },
                            "id": {
                                "type": ["null", "string"]
                            },
                            "value": {
                                "type": ["null", "string"]
                            }
                        }
                    }
                },
                {
                    "type": "null"
                }
            ]
        }
```

# Manual QA steps
Ran 

* discover
* check-tap
```
Checking stdin for valid Singer-formatted data
The output is valid.
It contained 458 messages for 19 streams.

     31 schema messages
    366 record messages
     61 state messages

Details by stream:
+----------------------+---------+---------+
| stream               | records | schemas |
+----------------------+---------+---------+
| communications       | 0       | 1       |
| clients              | 2       | 1       |
| deposit_products     | 1       | 1       |
| loan_products        | 1       | 1       |
| gl_accounts          | 1       | 5       |
| deposit_transactions | 29      | 1       |
| gl_journal_entries   | 263     | 1       |
| branches             | 1       | 1       |
| deposit_accounts     | 3       | 1       |
| cards                | 1       | 3       |
| credit_arrangements  | 0       | 1       |
| tasks                | 1       | 1       |
| custom_field_sets    | 21      | 1       |
| loan_accounts        | 1       | 1       |
| loan_repayments      | 35      | 7       |
| users                | 1       | 1       |
| centres              | 2       | 1       |
| loan_transactions    | 3       | 1       |
| groups               | 0       | 1       |
+----------------------+---------+---------+
```
* target-stitch

 
# Risks
 - 
 
# Rollback steps
 - revert this branch
